From cbed74938c81a25b5c1877420616c237721ef841 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Fri, 8 Sep 2017 19:58:22 -0500
Subject: [PATCH 2762/5242] MLK-16347-5: gpu: imx: dpu: TCON adjustment

commit  d6024e854b8f2d2e67906d191b673a36cf637e69 from
https://source.codeaurora.org/external/imx/linux-imx.git

The DPU TCON register regarding hsync/vsync are incorrectly
initialized. So, adjust vsync start to align with hsync start.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-tcon.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-tcon.c b/drivers/gpu/imx/dpu/dpu-tcon.c
index ff3fda0..eb76df3 100644
--- a/drivers/gpu/imx/dpu/dpu-tcon.c
+++ b/drivers/gpu/imx/dpu/dpu-tcon.c
@@ -130,11 +130,11 @@ void tcon_cfg_videomode(struct dpu_tcon *tcon, struct drm_display_mode *m)
 	dpu_tcon_write(tcon, 0x1, SMXFCTTABLE(0));
 
 	/* dsp_control[1]: vsync */
-	dpu_tcon_write(tcon, X(m->hsync_end) | Y(m->vsync_start - 1),
+	dpu_tcon_write(tcon, X(m->hsync_start) | Y(m->vsync_start - 1),
 								SPGPOSON(1));
 	dpu_tcon_write(tcon, 0x0, SPGMASKON(1));
 
-	dpu_tcon_write(tcon, X(m->hsync_end) | Y(m->vsync_end - 1),
+	dpu_tcon_write(tcon, X(m->hsync_start) | Y(m->vsync_end - 1),
 								SPGPOSOFF(1));
 	dpu_tcon_write(tcon, 0x0, SPGMASKOFF(1));
 
-- 
1.7.9.5

